File: Mobile/admin/lib/data/models/group.dart
//
//
//


import 'package:admin/data/models/student.dart';

class Group {
  int? groupId;
  int? doctorId;
  int? adminStudentId;
  int? projectId;
  List<Student> students  = [];

  Group({this.groupId, this.doctorId, this.adminStudentId, this.projectId});

  Group.fromJson(Map<String, dynamic> json) {
    groupId = json["group_id"];
    doctorId = json["doctor_id"];
    adminStudentId = json["admin_student_id"];
    projectId = json["project_id"];
    List<dynamic> arr = json["students"] ;
    for(int i = 0 ; i < arr.length ; i++) {
      students.add(Student.fromJson(arr[i])) ;
    }
  }

  Map<String, dynamic> toJson() {
    final group = <String, dynamic>{};
    if (groupId != null) {
      group["group_id"] = groupId;
    }
    if (doctorId != null) {
      group["doctor_id"] = doctorId;
    }
    if (adminStudentId != null) {
      group["admin_student_id"] = adminStudentId;
    }
    if (projectId != null) {
      group["project_id"] = projectId;
    }

    return group;
  }
}

File: projects_manager_backend/app/Models/Group.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Group extends Model
{
    use HasFactory;


    //
    protected $primaryKey = 'group_id' ;


    //
    public $timestamps = false;


    
    //
    protected $fillable = [
        'student_id' , 
        'doctor_id' ,
        'project_id' 
    ];




    // Group has many students belong to it 
    public function students()  : HasMany{
        return $this->hasMany(Student::class , 'group_id') ;
        
    }

    
}
